community/kernel/src/main/java/org/neo4j/kernel/impl/storemigration/UpgradableDatabase.java

```diff
@@ -84,6 +87,13 @@ public RecordFormats checkUpgradeable( File storeDirectory )
             return format;
         }
 
+        // If we are trying to open an enterprise store when configured to use community format, then inform the user
+        // of the config setting to change since downgrades aren't possible but the store can still be opened.
+        if ( result.actualVersion.startsWith( "vE" ) && !format.storeVersion().startsWith( "vE" ) )
```

Comment:
Imo it is better to keep this check in `UpgradableDatabase` because `RecordFormatSelector` should just select format and not care about upgrade paths we support.

The main thing I do not like about this solution is `startsWith( "vE" )`. What do you think about introducing a class with store version constants in community like:

```
class StoreVersions {
    private static final String STANDARD_V2_0 = "v0.A.1";
    // ...
    private static final String STANDARD_V3_0 = "v0.A.7";
    private static final String HIGH_LIMIT = "vE.H.0";
    
    private StoreVersions() {}

    public static boolean isCommunityStoreVersion(String storeVersion) {
         // assert that storeVersion among those we know
         return STANDARD_V2_0.equals(storeVersion) || STANDARD_V3_0.equals(storeVersion);
    }

     public static boolean isEnterpriseStoreVersion(String storeVersion) {
         // assert that storeVersion among those we know
        return HIGH_LIMIT.equals(storeVersion);
     }
}
```

Then all `RecordFormats` implementors would reference these constants:

```
public class HighLimit extends BaseRecordFormats {
    // ...
    public static final String STORE_VERSION = StoreVersions.HIGH_LIMIT;
    // ...
}
```

and in `UpgradableDatabase` check would look like:

```
 if ( isEnterpriseStoreVersion(result.actualVersion) && isCommunityStoreVersion(format.storeVersion()) ) {...}
```

with this approach knowledge about store versions is centralized. Drawback, of course, is that we put constant related to enterprise-only format in community. We already do this though in description of `GraphDatabaseSettings#record_format`.